### PR TITLE
Add custom rule to require template strings when concatting.

### LINKED ISSUE #9: ES2015 Strings: Disallow manual concatenation.
```javascript
// bad
let foo = "blah" + bar;
someFunction("foo '" + baz + "'");

// good
let foo = `blah${bar}`;
someFunction(`foo '${baz}`);
```

### PR DESCRIPTION
**Pull request by rwjblue**
Closes #9.

### CODE REVIEW

File: lib/jscsrc.json [hunk 1]
```diff
@@ -70,5 +70,6 @@
     "mark": "'",
     "escape": true
   },
-  "disallowVar": true
+  "disallowVar": true,
+  "requireTemplateStringsForConcatenation": true
```
Comment (brzpegasus) [inline lib/jscsrc.json]: Would be nice to alphabetize this.

Comment (rwjblue) [inline lib/jscsrc.json]: Agreed.  Will do.

**Comment by rwjblue** [comment]
Pushed another commit with better rule ordering.  Should be ready for another review.

**Comment by brzpegasus** [comment]
Another slight nitpick, but maybe we should rename the filename in `lib/rules` to match the actual rule name, so `lib/rules/require-template-strings-for-concatenation.js` instead of `lib/rules/require-template-strings.js`? The rest looks good!

**Comment by hzoo** [comment]
Ah cool - I was planning on adding something like this in jscs itself (and some other es6 rules)? https://github.com/jscs-dev/node-jscs/pull/1335. It's probably faster for you guys to just make your own rules but feel free to make an issue for any other requests.

**Comment by rwjblue** [comment]
@hzoo - Would love to move the custom rules upstream.

**Comment by hzoo** [comment]
Great I can help with that. I'm sure a lot of people are excited to style check ES6! Also I'll be creating https://github.com/jscs-dev/babel-jscs (issue https://github.com/jscs-dev/node-jscs/issues/1353) once I figure out how to fix more of the rules since `esprima-fb` will be deprecated (and you guys are using babel already).

**Pull request merged**